src/Temporalio/Worker/WorkflowInstance.cs

```diff
@@ -278,7 +332,10 @@ public WorkflowActivationCompletion Activate(WorkflowActivation act)
                         i++;
                     }
                 }
-                return completion;
+                // Unset the completion
```

Comment:
Obviously not for this PR but I noticed you don't do this in Python.

Reply:
That's because in python I considered it much less likely people would add commands outside of workflow execution. It's much easier to screw up and do it in .NET where we can't force the scheduler on you as well. But I can add it there, sure.